Sources/Client.swift

```diff
@@ -0,0 +1,334 @@
+    ///   - completion: Response with string representing transaction hash, or error.
+    /// - Throws: client error.
+    public func eth_sendTransaction(url: WCURL,
+                                    transaction: Transaction,
+                                    completion: @escaping RequestResponse) throws {
+        try handleTransaction(url: url, method: "eth_sendTransaction", transaction: transaction, completion: completion)
+    }
+
+    /// https://docs.walletconnect.org/json-rpc/ethereum#eth_signtransaction
+    /// Request to sign a transaction.
+    ///
+    /// - Parameters:
+    ///   - url: WalletConnect url object.
+    ///   - transaction: Transaction object.
+    ///   - completion: Response with string representing transaction signature, or error.
+    /// - Throws: client error.
+    public func eth_signTransaction(url: WCURL,
+                                    transaction: Transaction,
+                                    completion: @escaping RequestResponse) throws {
+        try handleTransaction(url: url, method: "eth_signTransaction", transaction: transaction, completion: completion)
+    }
+
+    private func handleTransaction(url: WCURL,
+                                   method: String,
+                                   transaction: Transaction,
+                                   completion: @escaping RequestResponse) throws {
+        let requestParamsData = try JSONEncoder().encode([transaction])
+        let requestParams = try JSONDecoder().decode(JSONRPC_2_0.Request.Params.self, from: requestParamsData)
+        let payload = JSONRPC_2_0.Request(method: method,
+                                          params: requestParams,
+                                          id: .string(UUID().uuidString))
+        let request = Request(payload: payload, url: url)
+        try send(request, completion: completion)
+    }
+
+    /// https://docs.walletconnect.org/json-rpc/ethereum#eth_sendrawtransaction
+    /// Request to send a raw transaction. Creates new message call transaction or
+    /// a contract creation for signed transactions.
+    ///
+    /// - Parameters:
+    ///   - url: WalletConnect url object.
+    ///   - data: Data as String.
+    ///   - completion: Response with the transaction hash, or the zero hash if the transaction is not
+    ///                 yet available, or error.
+    /// - Throws: client error.
+    public func eth_sendRawTransaction(url: WCURL, data: String, completion: @escaping RequestResponse) throws {
+        let payload = JSONRPC_2_0.Request(method: "eth_sendRawTransaction",
+                                          params: .positional([.string(data)]),
+                                          id: .string(UUID().uuidString))
+        let request = Request(payload: payload, url: url)
+        try send(request, completion: completion)
+    }
+
+    override func onConnect(to url: WCURL) {
+        print("WC: client didConnect url: \(url.bridgeURL.absoluteString)")
```

Comment:
I would exclude prints from the production code, or allow injecting a logger object

Reply:
added an issue